extensions/notebook/src/book/bookTreeItem.ts

```diff
@@ -161,6 +161,24 @@ export class BookTreeItem extends vscode.TreeItem {
 		return this.findChildSectionRecur(this, url);
 	}
 
+	/**
+	 * Determines whether a notebook is part of this book.
+	 * @param notebookUri Notebook uri Full URI of the notebook.
+	 */
+	public hasNotebook(notebookUri: string): boolean {
```

Comment:
I don't think I agree that this is the best place for this. The tree item should be just about the actual tree itself and not necessarily know anything about the contents of the book (only that it may have child book tree items)

Also we already have a getNotebook method on that model. Why can't you just use that? [https://github.com/microsoft/azuredatastudio/blob/master/extensions/notebook/src/book/bookModel.ts#L50](https://github.com/microsoft/azuredatastudio/blob/master/extensions/notebook/src/book/bookModel.ts#L50)

Reply:
# Resolved